packages/dds/tree/src/test/feature-libraries/forest-summary/incrementalSummary.spec.ts

```diff
@@ -0,0 +1,437 @@
+		if (current === undefined) {
+			break;
+		}
+		for (const [key, summaryObject] of Object.entries(current.tree)) {
+			if (summaryObject.type === SummaryType.Tree) {
+				if (key === forestSummaryKey) {
+					return summaryObject;
+				}
+				// Add to queue for BFS traversal
+				queue.push(summaryObject);
+			}
+		}
+	}
+	return undefined;
+}
+
+describe("Forest incremental summary", () => {
+	let factory: IChannelFactory;
+	let tree1: ISharedTree;
+	let view1: TreeView<typeof Board>;
+	let dataStoreRuntime1: MockFluidDataStoreRuntime;
+
+	// Stores the information about fields that should be incrementally encoded.
+	const incrementalFieldsMap: NestedMap<TreeNodeSchemaIdentifier, FieldKey, boolean> =
+		new Map();
+	const shouldEncodeFieldIncrementally = (
+		nodeIdentifier: TreeNodeSchemaIdentifier,
+		fieldKey: FieldKey,
+	): boolean => {
+		return tryGetFromNestedMap(incrementalFieldsMap, nodeIdentifier, fieldKey) ?? false;
+	};
+
+	before(() => {
+		// These are the fields that should be incrementally encoded.
+		setInNestedMap(
+			incrementalFieldsMap,
+			brand<TreeNodeSchemaIdentifier>(NoteList.identifier),
+			brand<FieldKey>("notes"),
+			true,
+		);
+		setInNestedMap(
+			incrementalFieldsMap,
+			brand<TreeNodeSchemaIdentifier>(Note.identifier),
+			brand<FieldKey>("label"),
+			true,
+		);
+		setInNestedMap(
+			incrementalFieldsMap,
+			brand<TreeNodeSchemaIdentifier>(NoteLabel.identifier),
+			brand<FieldKey>("labelText"),
+			true,
+		);
+	});
+
+	beforeEach(() => {
+		const options: SharedTreeOptionsInternal = {
+			jsonValidator: typeboxValidator,
+			forest: ForestTypeOptimized,
+			shouldEncodeFieldIncrementally,
+			treeEncodeType: TreeCompressionStrategyExtended.CompressedIncremental,
+		};
+		factory = configuredSharedTree(options as SharedTreeOptions).getFactory();
+
+		dataStoreRuntime1 = new MockFluidDataStoreRuntime({
+			clientId: `test-client`,
+			id: "test",
+			idCompressor: testIdCompressor,
+		});
+		tree1 = factory.create(dataStoreRuntime1, "TestSharedTree") as ISharedTree;
+
+		view1 = tree1.viewWith(
+			new TreeViewConfiguration({
+				schema: Board,
+			}),
+		);
+
+		view1.initialize(createInitialBoard(3, 2));
+
+		const forest = tree1.kernel.checkout.forest;
+		assert(forest instanceof ChunkedForest);
+	});
+
+	async function loadTreeView(treeNumber: number, summary: ISummaryTreeWithStats) {
+		const dataStoreRuntime = new MockFluidDataStoreRuntime({
+			clientId: `test-client-${treeNumber}`,
+			id: "test",
+			idCompressor: testIdCompressor,
+		});
+		const tree = (await factory.load(
+			dataStoreRuntime,
+			`TestSharedTree${treeNumber}`,
+			{
+				deltaConnection: dataStoreRuntime.createDeltaConnection(),
+				objectStorage: MockStorage.createFromSummary(summary.summary),
+			},
+			factory.attributes,
+		)) as ISharedTree;
+		return tree.viewWith(
+			new TreeViewConfiguration({
+				schema: Board,
+			}),
+		);
+	}
+
+	const testWithSuccessfulSummaries = async () => {
```

Comment:
From first look, its very hard to tell what functions like this do.

Its much easier to understand code if it pass in the data that a function uses as parameters rather than relying on what basically amounts to mutable globals.

Code like this is very hard to maintain/refactor/reuse/read.

If I want to add a new test to this file which tests something slightly different, maybe an optional instead of required field, or a type union with multiple allowed types, its very unclear how or if this code could be reused or adjusted without imapcting other tests which should not need to change.

Reply:
If there is a well defined thing which this function is doing, at least include a comment explaining what its for so people in the future can figure out if they should make their own different thing to test other cases, or extend this, or do something else entirly.

Reply:
Good point about adding a comment to functions like these that are very specific. I changed the logic for these tests so it doesn't apply anymore.